src/coreComponents/fieldSpecification/SourceFluxStatistics.cpp

```diff
@@ -0,0 +1,361 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file SourceFluxStatistics.cpp
+ */
+
+#include "SourceFluxStatistics.hpp"
+
+#include "SourceFluxBoundaryCondition.hpp"
+#include "FieldSpecificationManager.hpp"
+#include "LvArray/src/tensorOps.hpp"
+
+namespace geos
+{
+using namespace dataRepository;
+
+SourceFluxStatsAggregator::SourceFluxStatsAggregator( const string & name,
+                                                      Group * const parent ):
+  Base( name, parent )
+{
+  getWrapper< integer >( Group::viewKeyStruct::logLevelString() ).
+    appendDescription( GEOS_FMT( "\n- Log Level 1 outputs the sum of all {0}(s) produced rate & mass,\n"
+                                 "- Log Level 2 details values for each {0},\n"
+                                 "- Log Level 3 details values for each region.",
+                                 SourceFluxBoundaryCondition::catalogName() ) );
+
+  registerWrapper( viewKeyStruct::fluxNamesString().data(), &m_fluxNames ).
+    setRTTypeName( rtTypes::CustomTypes::groupNameRefArray ).
+    setInputFlag( InputFlags::REQUIRED ).
+    setSizedFromParent( 0 ).
+    setDescription( GEOS_FMT( "Name(s) array of the {0}(s) for which we want the statistics. "
+                              "Use \"all\" to target all {0}.",
+                              SourceFluxBoundaryCondition::catalogName() ) );
+}
+
+void SourceFluxStatsAggregator::postProcessInput()
+{
+  Base::postProcessInput();
+
+  FieldSpecificationManager & fsManager = FieldSpecificationManager::getInstance();
+  if( m_fluxNames.size() == 1 && m_fluxNames[0] == "all" )
```

Comment:
```suggestion
  if( m_fluxNames.size() == 1 && m_fluxNames[0] == "*" )
```
Even if we're not doing the `fnmatch` in this PR, would using `*` make sense?

Reply:
I agree, these `"all"` proposals came to my mind as it is possible to use that as `setNames` in field specifications.